scripts/components/api-changes-validator/api_changes_validator.ts

```diff
@@ -0,0 +1,93 @@
+import path from 'path';
+import { existsSync } from 'fs';
+import fsp from 'fs/promises';
+import { execa } from 'execa';
+import { ApiUsageGenerator } from './api_usage_generator.js';
+
+/**
+ * Validates changes between two versions of a package.
+ */
+export class ApiChangesValidator {
+  private readonly projectPath: string;
+  /**
+   * creates api changes validator
+   */
+  constructor(
+    private readonly packageName: string,
+    private readonly baselinePackageApiReportPath: string,
+    private readonly latestPackagePath: string,
+    private readonly workingDirectory: string
+  ) {
+    this.projectPath = path.join(workingDirectory, this.packageName);
+  }
+
+  validate = async (): Promise<void> => {
+    await fsp.rm(this.projectPath, { recursive: true, force: true });
+    await fsp.mkdir(this.projectPath, { recursive: true });
+    await this.createTestProject();
+    await execa('npx', ['tsc', '--build'], {
+      cwd: this.projectPath,
+      stdio: 'inherit',
+    });
+  };
+
+  private createTestProject = async () => {
+    const dependencies: Record<string, string> = {};
+    dependencies[
+      `${this.packageName}-latest`
+    ] = `file://${this.latestPackagePath}`;
```

Comment:
I'm planing to try verdaccio instead. Since it turned out we have to do local publish for different reasons. 
If that won't work well I'll try npm link.

Reply:
I ended up offering two options: using remote and using npm link.
remote is used by script, link is used in integration tests.
should be easy to modify this if needed.